docs/workflows/publish-host.rst

```diff
@@ -0,0 +1,92 @@
+Publish and Host
+================
+
+This section assumes that you have a repository, with content in it. To do this, see the
+:doc:`sync` or :doc:`upload` documentation.
+
+Create a Publisher
+------------------
+
+Publishers contain extra settings for how to publish. You can use a Python publisher on any
+repository that contains Python content.::
+
+$ http POST $PULP_BASE/pulp/api/v3/publishers/python/ name=bar
+
+Response::
+
+    {
+        "_href": "http://localhost:8000/pulp/api/v3/repositories/foo/publishers/python/bar/",
+        ...
+    }
+
+Create a variable for convenience.::
+
+$ export PUBLISHER_HREF=$(http $PULP_BASE/pulp/api/v3/publishers/python/ | jq -r '.results[] | select(.name == "bar") | ._href')
+
+
+Publish a repository with a publisher
+-------------------------------------
+
+Use the remote object to kick off a publish task by specifying the repository version to publish.
+Alternatively, you can specify repository, which will publish the latest version.
+
+The result of a publish is a publication, which contains all the information needed for ``pip`` to
+use. Publications are not consumable until they are hosted by a distribution.::
+
+$ http POST $PUBLISHER_HREF'publish/' repository=$REPO_HREF
+
+Response::
+
+    [
+        {
+            "_href": "http://localhost:8000/pulp/api/v3/tasks/fd4cbecd-6c6a-4197-9cbe-4e45b0516309/",
+            "task_id": "fd4cbecd-6c6a-4197-9cbe-4e45b0516309"
+        }
+    ]
+
+Create a variable for convenience.::
+
+$ export PUBLICATION_HREF=$(http $PULP_BASE/pulp/api/v3/publications/ | jq -r --arg PUBLISHER_HREF "$PUBLISHER_HREF" '.results[] | select(.publisher==$PUBLISHER_HREF) | ._href')
+
+Host a Publication (Create a Distribution)
+--------------------------------------------
+
+To host a publication, (which makes it consumable by ``pip``), users create a distribution which
+will serve the associated publication in the content app, namespaced by ``base_path``.::
+
+$ http POST $PULP_BASE/pulp/api/v3/distributions/ name='baz' base_path='foo' publication=$PUBLICATION_HREF
+
+Response::
+
+    {
+        "_href": "http://localhost:8000/pulp/api/v3/distributions/9b29f1b2-6726-40a2-988a-273d3f009a41/",
+       ...
+    }
+
+
+Use the newly created distribution
+-----------------------------------
+
+The metadata and packages can now be retrieved from the distribution::
+
+$ curl localhost:8000/pulp/content/foo/simple/
+$ curl localhost:8000/pulp/content/foo/simple/shelf-reader/
+
+The content is also pip installable.::
+
+$ pip install --trusted-host dev.pulp3 -i http://dev.pulp3:8000/pulp/content/foo/simple/ shelf-reader
```

Comment:
Don't use a hostname specific to our vagrant environment (and I think this change hasn't even happened yet). If we use localhost everywhere else we should probably use it here.